app/src/main/java/org/gdg/frisbee/android/view/LocationListPreference.java

```diff
@@ -0,0 +1,295 @@
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.preference.DialogPreference;
+import android.preference.PreferenceManager;
+import android.text.Editable;
+import android.text.TextUtils;
+import android.text.TextWatcher;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.Adapter;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.Filter;
+import android.widget.ListView;
+
+import org.gdg.frisbee.android.R;
+
+public class LocationListPreference extends DialogPreference {
+
+    private FilterListView listView;
+    private int clickedItemIndex;
+    private String mValue;
+    private boolean mValueSet;
+    private String[] mEntries;
+    private String[] mEntryValues;
+
+    public LocationListPreference(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        setDialogLayoutResource(R.layout.view_location_list_preference);
+    }
+
+    @Override
+    protected void onPrepareDialogBuilder(AlertDialog.Builder builder) {
+        super.onPrepareDialogBuilder(builder);
+        builder.setPositiveButton(null, null);
+    }
+
+    @Override
+    protected void onBindDialogView(View view) {
+        super.onBindDialogView(view);
+        listView = (FilterListView) view.findViewById(android.R.id.list);
+        clickedItemIndex = findIndexOfValue(mValue);
+        listView.setOnItemClickListener(
+                new AdapterView.OnItemClickListener() {
+                    @Override
+                    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                        clickedItemIndex = findIndexByLabel((String) parent.getItemAtPosition(position));
+                        LocationListPreference.this.onClick(getDialog(), DialogInterface.BUTTON_POSITIVE);
+                        getDialog().dismiss();
+                    }
+                }
+        );
+        listView.setChoiceMode(ListView.CHOICE_MODE_SINGLE);
+        CheckedItemAdapter adapter = new CheckedItemAdapter(
+                getContext(), android.R.layout.simple_list_item_single_choice,
+                android.R.id.text1, getEntries()
+        );
+        listView.setAdapter(adapter);
+        clickedItemIndex = findIndexOfValue(getPersistedString(null));
+        listView.setSelection(clickedItemIndex);
+        listView.setItemChecked(clickedItemIndex, true);
+        listView.setTextFilterEnabled(true);
+        listView.setFilterListener(
+                new Filter.FilterListener() {
+                    @Override
+                    public void onFilterComplete(int count) {
+                        int index = findIndexByLabelInFilteredListView(getPersistedString(null));
+                        listView.setItemChecked(index, true);
+                    }
+                }
+        );
+
+        EditText editText = (EditText) view.findViewById(R.id.filter);
```

Comment:
I would use a `SearchView` here to have beautiful search icon next to the texts.